x-pack/plugins/ml/public/application/services/anomaly_timeline_service.ts

```diff
@@ -49,6 +50,14 @@ export class AnomalyTimelineService {
     this.timeFilter.enableTimeRangeSelector();
   }
 
+  public static isSwimlaneData(arg: unknown): arg is SwimlaneData {
+    return isPopulatedObject(arg, ['interval', 'points', 'laneLabels']);
+  }
+
+  public static isOverallSwimlaneData(arg: unknown): arg is OverallSwimlaneData {
+    return this.isSwimlaneData(arg) && isPopulatedObject(arg, ['earliest', 'latest']);
+  }
```

Comment:
I just realized this typguard is not going to work as expected. `ViewBySwimLaneData` extends `OverallSwimlaneData`, hence it might return true for the View by swim lane as well.

Reply:
Enforced a more strict check here https://github.com/elastic/kibana/pull/105696/commits/46311d5764593698a52e8e402d7137d312428b8d